File: 05_Heaps/MaxHeapsC++/Driver.cpp
#include "MaxHeapsCPlusPlus.hpp"

int main()
{
	initializeList(20);
	cout<<"Original Array\n";
	printList();
	
	cout<<"Build Max Heap\n";
	BuildMaxHeap();
	printList();
	
	cout<<"Heap Sort\n";
	HeapSort();
	printList();
	
	cout<<"Extract Max\n";
	cout<<extractMax()<<endl;
	
	cout<<"Increase Key\n";
	increaseKey(8, 87);
	printList();
	
	cout<<"Insert Value\n";
	insertVal(16);
	printList();
	
	cout<<"Heap Sort\n";
	HeapSort();
	printList();
}

File: 05_Heaps/Max Heaps/MaxHeapsPython/driver.py
from MaxHeapsList import *
def main():
        initializeList(20)
        print("Original List")
        print(myList)
        
        print("Build Max Heap")
        BuildMaxHeap()
        print(myList)
        
        print("Heap Sort")
        HeapSort()
        print(myList)
        
        print("Extract Max")
        print(extractMax())
        
        print("Increase key")
        increaseKey(8, 87)
        print(myList)
        
        print("Insert Value")
        insertVal(16)
        print(myList)
        
        print("Heap Sort")
        HeapSort()
        print(myList)

if __name__ == "__main__":
        main()